text/0000-deprecation-ember-map.md

```diff
@@ -0,0 +1,37 @@
+- Start Date: 2017-07-20
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the deprecation of the following classes:
+
+- `Ember.OrderedSet`
+- `Ember.Map`
+- `Ember.MapWithDefault`
+
+These classes need to be moved to an external addon given they are private classes and unused in Ember.js itself.
+
+# Motivation
+
+These classes have not been used in Ember itself for a while now. They have always been private but they are used in a few addons, and in particular Ember Data is using them.
+
+# Transition Path
+
+The classes will be extracted to an addon and the ones in Ember will be deprecated.
```

Comment:
If that is in the scope of this RFC, it might be worthwhile to take a look at switching to the native ES6 `Map`. The `MapWithDefault` behavior can be easily achieved: [Babel REPL example](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=true&presets=es2016%2Ces2017%2Cstage-0&targets=&browsers=&builtIns=false&debug=false&code_lz=MYewdgzgLgBA-gEwKYDMCGBXANlAamrDJAMRACcYBeGAZQE8BbAIxCwAoByZJ9bPAohwCUAbgBQY4FjQQIMALJoADgHUAllAAWAEVSYcMJAA8oSMAjmKlMAN5iYMUJChkMwKOTbJeOfIRLkADQwGkhkaExYSEK29g4hKDBsUHRKSCCJ3vr8_qQUAISU1BwoGGDuauDCsfHxWmQgAO4wYEjNACqpSACiZA1knACaIBgwDBjQMEoNAG5qyDBoMFl8fkR5MKXlUJVgMB4Kyupauj5QwuK1AL5xDhAYaQOh4ZHRl3WaahAA2oh6qwIAmQALpUZb_XyAvLvG4OOIAcyQUDYAGskHQYnZampEslPhAAHSaGSo9FCTG3eJkJEYMh7e6PAmI5FojHvByw-JOSYzQFgrRfX4rSG5cjA0lsykCwkQJES4K8_yiSnUqC0vaKogwsQ3STgSYMZRg1rNKzHHQQlnoqgAPiSNhgrOCwGpaFMCAAglAAFwtNowbRupAwK7k8R6yCsJAErAgeFsQ1KJlykogEDCZXcqMxuMJ5QEswuNRICBsMNAA)

I would be available for such a PR in Ember Data. :raising_hand_man: